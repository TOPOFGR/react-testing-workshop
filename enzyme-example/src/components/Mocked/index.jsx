import React from 'react';
import PropTypes from 'prop-types';

const Mocked = ({ onSubmit }) => {

    console.log('API call');

    const onSubmitHandler = (e) => {
        e.preventDefault();
    
        onSubmit();
    };

    return (
        <div>
            Mocked component

            <form onSubmit={onSubmitHandler}>
                <input type="text" />
            <button>Save</button>
            </form>
        </div>
    );
};

Mocked.propTypes = {
    onSubmit: PropTypes.func.isRequired,
};

export default Mocked;
