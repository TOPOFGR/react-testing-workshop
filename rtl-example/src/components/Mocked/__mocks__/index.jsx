import React from 'react';

const Mocked = ({onSubmit}) => {
    return (
        <>
         <button onClick={onSubmit} type="button">Click</button>   
        </>
    );
};

export default Mocked;