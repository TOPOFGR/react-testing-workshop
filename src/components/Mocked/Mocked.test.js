import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import Mocked from ".";

describe('<Mocked/>', () => { 

    beforeEach(() => {
        jest.resetAllMocks();
    });
    
    it('should render without crashing', () => { 
        const onSubmit = jest.fn();

        const { asFragment } = render(<Mocked onSubmit={onSubmit}/>);


        expect(asFragment()).toMatchSnapshot();

     });
    
    it('should render the correct text', () => { 
        const onSubmit = jest.fn();

        render(<Mocked onSubmit={onSubmit}/>);

        const title = screen.getByText('Mocked component');

        expect(title).toBeInTheDocument();

     });
    
    it('should call the onSubmit function', () => { 
        const onSubmit = jest.fn();

        render(<Mocked onSubmit={onSubmit}/>);

        const input = screen.getByRole('textbox');
        
        userEvent.type(input, 'Hello');
        
        const button = screen.getByRole('button', { name: /Save/i });

        userEvent.click(button);

        expect(onSubmit).toHaveBeenCalledTimes(1);

     });
 })