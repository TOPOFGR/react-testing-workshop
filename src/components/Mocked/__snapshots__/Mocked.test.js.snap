// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Mocked/> should render without crashing 1`] = `
<DocumentFragment>
  <div>
    Mocked component
    <form>
      <input
        type="text"
      />
      <button>
        Save
      </button>
    </form>
  </div>
</DocumentFragment>
`;
